repo-validation/infrastructure/file_repository_test.go

```
@@ -0,0 +1,357 @@
+				WithTemplateService(template.NewService(&config.Config{})),
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			repo := NewFileRepository(tt.path, tt.options...)
+			if repo == nil {
+				t.Errorf("NewFileRepository() returned nil")
+			}
+			if repo.basePath != tt.path {
```

golangci-lint (1.64.8)

SA5011: possible nil pointer dereference

(staticcheck)